### PR TITLE
feat: add style/className/styles/classNames to ConfigProvider

### PR DESCRIPTION
**Pull request by LuZhenJie1999**
### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/discussions/39862

### 💡 Background and solution
add style/styles/className/classNames to ConfigProvider for button

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸  English | add style/styles/className/classNames to ConfigProvider for button          |
| 🇨🇳 Chinese | ConfigProvider 可自定义 button 的 style/className/styles/classNames           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 32c90ac</samp>

This pull request adds a new feature to the `ConfigProvider` component, which allows users to customize the style and className props of the `Button` component globally. It updates the `Button` component, the `ConfigContext` interfaces, and the documentation to support this feature. It also adds a test file and fixes a minor issue in the `Button` component.

### <samp>🤖 Generated by Copilot at 32c90ac</samp>

*  Add `button` prop to `ConfigProvider` component to allow customizing `Button` component style and className ([link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR37-R45), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR83), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R20), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R90), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R133), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R66), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R67))
*  Import and use `IconWrapper` component in `Button` component to wrap icon prop and apply styles and classes ([link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R21), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L241-R250), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L127-R127), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R265), [link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140R280))
*  Remove duplicate import of `IconWrapper` component in `Button` component ([link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L26))
*  Reorder imports and add new line in `context.ts` file for readability ([link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaL1-R8))
*  Add test cases for `ConfigProvider.button` feature in `button.test.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42623/files?diff=unified&w=0#diff-b8043a33d37205c067d3278ba56e3d1e4ccb36ac63fa46ec5f7775fa2e6a16a0R1-R53))

### CODE REVIEW

File: components/button/button.tsx [hunk 1]
```diff
@@ -271,6 +277,7 @@ const InternalButton: React.ForwardRefRenderFunction<
       {...(rest as NativeButtonProps)}
       type={htmlType}
       className={classes}
+      style={button?.style}
```
Comment (MadCcc) [inline components/button/button.tsx:280]: 要和用户传的 style mix 一下

File: components/config-provider/index.zh-CN.md [hunk 2]
```diff
@@ -64,6 +64,7 @@ export default Demo;
 | iconPrefixCls | 设置图标统一样式前缀 | string | `anticon` | 4.11.0 |
 | input | 设置 Input 组件的通用属性 | { autoComplete?: string } | - | 4.2.0 |
 | select | 设置 Select 组件的通用属性 | { showSearch?: boolean } | - |  |
+| button | 设置 Button 组件的 style 和 className 属性 | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - |  |
```
Comment (MadCcc) [inline components/config-provider/index.zh-CN.md:67]:
```suggestion
| button | 设置 Button 组件的 style 和 className 属性 | - | - |  |
```
这里可以留个链接去 Button 的页面

Comment (MadCcc) [inline components/config-provider/index.zh-CN.md:67]:
```suggestion
| button | 设置 Button 组件的通用属性 | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - |  |
```

**Comment by MadCcc** [comment]
size-limit 超了
https://github.com/ant-design/ant-design/pull/42554
等等这个合了

File: components/button/button.tsx [hunk 3]
```diff
@@ -238,11 +239,17 @@ const InternalButton: React.ForwardRefRenderFunction<
     compactItemClassnames,
     className,
     rootClassName,
+    button?.className,
   );
 
+  const fullStyle = { ...button?.style, ...customStyle };
+
+  const iconClasses = classNames(customClassNames?.icon, button?.classNames?.icon);
+  const iconStyle = { ...(styles?.icon || {}), ...(button?.styles?.icon || {}) };
```
Comment (MadCcc) [inline components/button/button.tsx:248]: 这个顺序要换一下，组件的 styles 优先级更高

File: components/config-provider/index.zh-CN.md [hunk 4]
```diff
@@ -64,6 +64,7 @@ export default Demo;
 | iconPrefixCls | 设置图标统一样式前缀 | string | `anticon` | 4.11.0 |
 | input | 设置 Input 组件的通用属性 | { autoComplete?: string } | - | 4.2.0 |
 | select | 设置 Select 组件的通用属性 | { showSearch?: boolean } | - |  |
+| button | 设置 Button 组件的通用属性 | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - |  |
```
Comment (li-jia-nan) [inline components/config-provider/index.zh-CN.md:67]:
```suggestion
| button | 设置 Button 组件的通用属性 | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - | 5.6.0 |
```

File: components/config-provider/index.en-US.md [hunk 5]
```diff
@@ -63,6 +63,7 @@ Some components use dynamic style to support wave effect. You can config `csp` p
 | iconPrefixCls | Set icon prefix className | string | `anticon` | 4.11.0 |
 | input | Set Input common props | { autoComplete?: string } | - | 4.2.0 |
 | select | Set Select common props | { showSearch?: boolean } | - |  |
+| button | Set Select common props | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - |  |
```
Comment (li-jia-nan) [inline components/config-provider/index.en-US.md:66]:
```suggestion
| button | Set Select common props | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - | 5.6.0 |
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**